src/ppx/geometry.cpp

```diff
@@ -54,22 +54,23 @@ class VertexDataProcessorBase
     // ----------------------------------------
 
     // Missing attributes will also result in NOOP.
-    // returns the element count of the vertex buffer with the bufferIndex
+    // Returns the number of bytes written, since partial elements can be written
     // 0 is returned if the bufferIndex is ignored
     template <typename U>
     uint32_t AppendDataToVertexBuffer(Geometry* pGeom, uint32_t bufferIndex, const U& data)
     {
         if (bufferIndex != PPX_VALUE_IGNORED) {
             PPX_ASSERT_MSG((bufferIndex >= 0) && (bufferIndex < pGeom->mVertexBuffers.size()), "buffer index is not valid");
+            uint32_t originalSize = pGeom->mVertexBuffers[bufferIndex].GetDataSize();
             pGeom->mVertexBuffers[bufferIndex].Append(data);
-            return pGeom->mVertexBuffers[bufferIndex].GetElementCount();
+            return pGeom->mVertexBuffers[bufferIndex].GetDataSize() - originalSize;
```

Comment:
in the case of `mMaxElementCount > 0`, and since we pre reserved the max size,  `GetDataSize` would always return the max size before & after the Append(), and this function here would return 0?

Reply:
Actually in the case of `mMaxElementCount > 0`, the behaviour of `GetDataSize` and `Append` are changed so that `mOffset` is used to determine what the true size of the data is, not the size of the data vector. So before and after `Append()`, `mOffset` will be different, and this line will not be 0.

Reply:
i see, make sense